### ISSUE TITLE
[question] `conan install -u` prefers upstream version

### ISSUE THREAD

**Issue by zulli73** [conan-io/conan#6544]
`conan install -u` seems to prefer versions from "the server" even if a newer version (according to semver) is available in the cache

From the docs,

> -u, --update          Check updates exist from upstream remotes

From this help text I would expect that it "checks" whether newer version exists upstream - but not that it ignores local versions. In my particular case, I have this version in the cache:
- 1.2.1-200213104101+0000000

And the following on the server:
- 1.2.1-200212075818+2424680

Both in `jenkins/testing`. I would expect the former to win. I even wrote a small python scrip to test `python-semver`'s behavior with these two versions and the former wins (see code below). This led me to the conclusion that conan completely ignores local version if `-u` is specified.

Assuming my obvservations are right, is this the desired behavior? And if so, how can I tweak this to behavior so that 'cache local' and 'upstream' versions are treated equally?

```py
from semver import max_satisfying, satisfies
range_ = '1.x'
print('Better:', max_satisfying(['1.2.1-200213104101+0000000', '1.2.1-200212075818+2424680'
], range_, loose=False, include_prerelease=True))
```

**Comment by jgsogo** [comment]
Hi, @zulli73!  

Even though the documented argument is pretty clear, I agree with you and I would expect it to return the latest version from the combined list of the cache and the server or, at least, we should provide a way to do it. As it is now, it is confusing, a call to `conan install ... --update` might install an older version than the one in the cache.

For these Conan 1.x versions we cannot change its behavior because we will break some CI flows of people relying on it, but I'll tag it for Conan v2.0 when we are going to rewrite all the CLI.

Thanks!

**Comment by zulli73** [comment]
Sounds great, thanks!

For those suffering from the same issue, the following 'trick' **seems** to workaround this issue:

```sh
# Downloads the latest version available upstream (if not
# already in the cache)
conan install -u --no-import ... 
# installs the latest version available in the cache (either
# a version that was already there before or the just
# downloaded version)
conan install ...
```

**Comment by memsharded** [comment]
This was solved some time ago in 2.0 betas, added a test to make it very explicit in https://github.com/conan-io/conan/pull/13102, but otherwise, this is solved in 2.0, to be released very soon.
